src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -0,0 +1,534 @@
+            }
+
+            private async Task<byte[]> GetResponseData()
+            {
+                if (_data != null)
+                {
+                    return _data;
+                }
+
+                using (System.Runtime.InteropServices.JavaScript.ArrayBuffer dataBuffer = (System.Runtime.InteropServices.JavaScript.ArrayBuffer)await _status.ArrayBuffer().ConfigureAwait(continueOnCapturedContext: true))
+                {
+                    using (Uint8Array dataBinView = new Uint8Array(dataBuffer))
+                    {
+                        _data = dataBinView.ToArray();
+                        _status.Dispose();
+                    }
+                }
+
+                return _data;
+            }
+
+            protected override async Task<Stream> CreateContentReadStreamAsync()
+            {
+                byte[] data = await GetResponseData().ConfigureAwait(continueOnCapturedContext: true);
+                return new MemoryStream(data, writable: false);
+            }
+
+            protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context) =>
+                SerializeToStreamAsync(stream, context, CancellationToken.None);
+            protected sealed override async Task SerializeToStreamAsync(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+            {
+                byte[] data = await GetResponseData().ConfigureAwait(continueOnCapturedContext: true);
+                await stream.WriteAsync(data, 0, data.Length, cancellationToken).ConfigureAwait(continueOnCapturedContext: true);
+            }
+            protected internal override bool TryComputeLength(out long length)
+            {
+                if (_data != null)
+                {
+                    length = _data.Length;
+                    return true;
+                }
+
+                length = 0;
+                return false;
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                _status?.Dispose();
+                base.Dispose(disposing);
+            }
+        }
+
+        private sealed class WasmHttpReadStream : Stream
+        {
+            private WasmFetchResponse? _status;
+            private JSObject? _reader;
+
+            private byte[]? _bufferedBytes;
+            private int _position;
+
+            public WasmHttpReadStream(WasmFetchResponse status)
+            {
+                _status = status;
+            }
+
+            public override bool CanRead => true;
+            public override bool CanSeek => false;
+            public override bool CanWrite => false;
+            public override long Length => throw new NotSupportedException();
+            public override long Position
+            {
+                get => throw new NotSupportedException();
+                set => throw new NotSupportedException();
+            }
+
```

Comment:
Nit: you might want to override BeginRead/EndRead as well; if you don't and someone uses them, they'll end up getting an exception about synchronous reads not being supported, as by default the base implementations just queue a work item that wraps the sync method.  You can implement these pretty easily with an internal helper that should already be available in this library, e.g. something like:
```C#
public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsynCallback? callback, object? state) =>
    TaskToApm.Begin(ReadAsync(buffer, offset, count), callback, state);

public override int EndRead(IAsyncResult asyncResult) => TaskToApm.End<int>(asyncResult);
```
Search for BeginRead/EndRead and you should find other examples in this project.